Zend/zend_object_handlers.c

```diff
@@ -169,7 +169,7 @@ ZEND_API HashTable *zend_std_get_debug_info(zend_object *object, int *is_temp) /
 		return ht;
 	}
 
-	zend_error_noreturn(E_ERROR, ZEND_DEBUGINFO_FUNC_NAME "() must return an array");
+	zend_error_noreturn(E_ERROR, "Method %s::__debugInfo() must return an ?array", ZSTR_VAL(ce->name));
```

Comment:
`Return value of %s::__debugInfo() must be of type ?array, %s returned` would be more in line with other return type error messages

Reply:
I don't remember having seen an error message with a nullable value represented like the syntax, is that something we do in other places? If not then I would prefer `... of type array or null...` (or whichever we error text we use elsewhere to be consistent)

Reply:
@KalleZ It is something we changed for PHP 8. We are now displaying nullable types everywhere like this.

Reply:
@kocsismate awesome then, no objections on my end then! Thanks

Reply:
@kocsismate Align with what messages, ZPP?

Because all the "magical" ones do not mention it:

- __serialize(): https://3v4l.org/HLiTj
- __toString(): https://3v4l.org/Dbe6G
- __debugInfo(): https://3v4l.org/0EEPh
- __sleep(): https://3v4l.org/dH96A

Also, I'd need to figure out how to populate `%s returned`. Maybe `Z_TYPE(retval)` does the job?

Reply:
@carusogabriel Yes, it's ZPP. I think the other messages should (but at least could) be converted to that format later.

You can use `zend_zval_type_name(retval)` to get the string representation of a zval. :)